python/ray/tune/schedulers/pb2_utils.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+from scipy.optimize import minimize
+
+import GPy
+from GPy.kern import Kern
+from GPy.core import Param
+from sklearn.metrics import pairwise_distances
+from sklearn.metrics.pairwise import euclidean_distances
+
+
+## PB2 time varying kernel
```

Comment:
Can you remove these comments for all functions in this file please and add docstring instead. For the simple ones just a one line docstring is fine, but for the more complicated ones it would be great to also describe what the args/return values are.

Reply:
Done.